usd/src/sdf_parser/Material.cc

```diff
@@ -0,0 +1,462 @@
+
+// TODO(ahcorde) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/base/tf/stringUtils.h>
+#include <pxr/usd/sdf/types.h>
+#include <pxr/usd/usd/prim.h>
+#include <pxr/usd/usdShade/connectableAPI.h>
+#include <pxr/usd/usdShade/materialBindingAPI.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/Pbr.hh"
+
+namespace sdf
+{
+// Inline bracket to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+  /// \brief Copy a file in a directory
+  /// \param[in] _path path where the copy will be located
+  /// \param[in] _fullPath name of the file to copy
+  /// \return True if the file at _fullPath was copied to _path. False otherwise
+  bool copyMaterial(const std::string &_path, const std::string &_fullPath)
+  {
+    if (!_path.empty() && !_fullPath.empty())
+    {
+      auto fileName = ignition::common::basename(_path);
+      auto filePathIndex = _path.rfind(fileName);
+      auto filePath = _path.substr(0, filePathIndex);
+      if (!filePath.empty())
+      {
+        ignition::common::createDirectories(filePath);
+      }
+      return ignition::common::copyFile(_fullPath, _path);
+    }
+    return false;
+  }
+
+  /// \brief Get the path to copy the material to
+  /// \param[in] _uri full path of the file to copy
+  /// \return A relative path to save the material. The path looks like:
+  /// materials/textures/<filename with extension>
+  std::string getMaterialCopyPath(const std::string &_uri)
+  {
+    return ignition::common::joinPaths(
+      "materials",
+      "textures",
+      ignition::common::basename(_uri));
+  }
+
+  /// \brief Fill Material shader attributes and properties
+  /// \param[in] _prim USD primitive
+  /// \param[in] _name Name of the field attribute or property
+  /// \param[in] _vType Type of the field
+  /// \param[in] _value Value of the field
+  /// \param[in] _customData Custom data to set the field
+  /// \param[in] _displayName Display name
+  /// \param[in] _displayGroup Display group
+  /// \param[in] _doc Documentation of the field
+  /// \param[in] _colorSpace if the material is a texture, we can specify the
+  /// color space of the image
+  template<typename T>
+  void CreateMaterialInput(
+    const pxr::UsdPrim &_prim,
+    const std::string &_name,
+    const pxr::SdfValueTypeName &_vType,
+    const T &_value,
+    const std::map<pxr::TfToken, pxr::VtValue> &_customData,
+    const pxr::TfToken &_displayName,
+    const pxr::TfToken &_displayGroup,
+    const std::string &_doc,
+    const pxr::TfToken &_colorSpace = pxr::TfToken(""))
+  {
+    auto shader = pxr::UsdShadeShader(_prim);
+    if (!shader)
+    {
+      std::cerr << "Not able to convert the prim to a UsdShadeShader\n";
```

Comment:
Instead of printing to `sdf::cerr`, can we create a `UsdError` object and return that? We will probably need to modify the `CreateMaterialInput` API to return a `UsdErrors` object instead of returning `void`.